tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -197,22 +196,31 @@ export function waitForAnyProcessOutputToMatch(
 }
 
 export async function killAllProcesses(signal = 'SIGTERM'): Promise<void> {
-  await Promise.all(
-    _processes.map(
-      ({ pid }) =>
-        new Promise<void>((resolve, reject) => {
-          treeKill(pid, signal, (err) => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve();
-            }
-          });
-        }),
-    ),
-  );
+  const processesToKill: Promise<void>[] = [];
+
+  while (_processes.length) {
+    const childProc = _processes.pop();
+    if (!childProc) {
```

Comment:
When would `childProc` not be set? I think it always would be since nothing in this loop is async...?

Reply:
`pop` can technically return undefined when there are no items in the array.

Reply:
Everything is synchronous until the `await` at the end of the method though so the `_processes.length` check is still valid.

Doesn't hurt, but I think this `if` is unnecessary 🤷